tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -823,10 +823,33 @@ def complex_agg(x, agg):
     res = [complex_agg(fft[config["coeff"]], config["attr"]) if config["coeff"] < len(fft)
            else np.NaN for config in param]
     index = ['coeff_{}__attr_"{}"'.format(config["coeff"], config["attr"]) for config in param]
+
+    # Spectral Centroid:
+    fft_abs = np.abs(fft)
+    res += [fft_abs[:len(x)//2].dot(np.arange(0, len(x)//2, 1)) / fft_abs[:len(x)//2].sum()]
+    index += ['centroid']
+
     return zip(index, res)
 
 
 @set_property("fctype", "simple")
+def spectral_centroid(x):
+    """
+    Returns the spectral centroid, which is the average frequency in the absolute fourier transform spectrum.
+
+    :param x: the time series to calculate the feature of
+    :type x: pandas.Series
+    :return: the value of this feature
+    :return type: float
+    """
+
+    fft_abs = abs(np.fft.rfft(x))
+
+    # Spectral centroid (weights average over abs values):
+    return fft_abs[:len(x)//2].dot(np.arange(0, len(x)//2, 1)) / fft_abs[:len(x)//2].sum()
```

Comment:
I interpret the `cc @nikhaas` from Max as a hinto to give my 2cents.

First of all, I see that you are only interested in "well behaved" frequencies, because you discard the last entry if `fft_abs` has an odd number of items. From the docs of [`np.fft.rfft`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.fft.rfft.html), we see that this last frequency is complex in general.

Your [reference from wikipedia](https://en.wikipedia.org/wiki/Spectral_centroid) seems only meaningful in the time domain. But as always with DFT, we carry no information about the time domain. I've also seen a [slightly different implementation](https://gist.github.com/endolith/359724/aa7fcc043776f16f126a0ccd12b599499509c3cc), where the author uses `np.arange(0,1,len(x)//2)` instead of `np.arange(0, len(x)//2, 1)`. But I prefer your implementation because of the association to the bin. Transforming back in the time domain is done by `val * seconds * sampling_freq / bin_no`.

However, I agree with Max: We could normalize all DFT values which yields a discrete probability distribution of the bins and then calculate some moments (if this is useful at all).

Reply:
> I interpret the cc @nikhaas from Max as a hinto to give my 2cents.

I cced you because you both seem to work on the same problem (classification of vibration signals)

> DFT, we carry no information about the time domain

Why? we assume that the signal is uniformly sampled for DFT right?

> However, I agree with Max: We could normalize all DFT values which yields a discrete probability distribution of the bins and then calculate some moments (if this is useful at all).

my intention was just to have a feature calculator that calculates transformations of the whole fft vector, which does not mean that it have to be normalized per se.

Reply:
> Why? we assume that the signal is uniformly sampled for DFT right?

That's true.

> my intention was just to have a feature calculator that calculates transformations of the whole fft vector, which does not mean that it have to be normalized per se.

This might need some discussion. I can think about relevant aggregates w/o normalization (like max or argmax) and w/ normalization (centroid, kurtosis)..
Ok then, thanks for the hint!